lib/graphql/transaction.ts

```diff
@@ -501,4 +521,9 @@ export class Transaction extends IterableModel<TransactionModel> {
     const result: Query = await this.client.rawQuery(FETCH_TRANSACTIONS_CSV, args);
     return result.viewer?.mainAccount?.transactionsCSV ?? "";
   }
+
+  public async fetchFilterPresets(): Promise<FilterPreset[]> {
```

Comment:
let's put all possible `FilterPreset` implementations as a union here, so that the client can differentiate them by the `__typename` field

```suggestion
  public async fetchFilterPresets(): Promise<(GenericFilterPreset | MissingTaxAssetsFilterPreset)[]> {
```